### ISSUE TITLE
Interface Compile Bug Across Projects in Workspace

### ISSUE THREAD

**Issue by ESS-BrianValdillez** [beefytech/Beef#1783]
Getting what I think is a compile bug when using interfaces across multiple projects. I have a simple interface, `ITaggable`, which is as such:

```
    // A simple interface to retrieve and match entities via tags.
    public interface ITaggable
    {
        public ABIString Tag
        {
            get;
        }

        public bool MatchesTag(ABIString tag)
        {
            return String.Matches(Tag, tag);
        }

        
    }
```

I have this interface in our engine's project. I then tried to apply this interface to our test class, Taggy,  as follows:
```
public class Taggy : ITaggable
	{
		public ABIString Tag
		{
			get => .("MyTag");
		}
	}
```	
in our executable's project, and got this error:

```
ERROR: Property is not implemented by this type
  while implementing default interface method 'Logic.Taggy.RSDK.ITaggable.MatchesTag(RSDK.ABIString tag)' at line 19:26 in c:\EveningStar\Project Yo\Submodules\StarEngine\Source\RSDKBeef\src\Interfaces\ITaggable.bf
   return String.Matches(Tag, tag);
```

If I move the Taggy class from the executable's project to the engine's project, the compile error goes away. My uneducated guess is that It might be something to do with the compiler checking the ability to compile `MatchesTag` for Taggy before it compiles the implemented Tag property on Taggy.

Version:
Beef IDE Version 0.43.5 (Nightly 09/14/2022) Build 70830358c803f0458da3276f25b0aca60fab17e3
(We are on an older version simply because we try to avoid updating unless there are specific fixes we need for the project's purpose).

**Comment by ESS-BrianValdillez** [comment]
I can't post a single copy-pasted file, but I did make a small sample project to demo. It has the interface, a compiling implementation within then Engine side, and then a non-compiling implementation on the Exe side. 

[BeefBug_1783.zip](https://github.com/beefytech/Beef/files/10405460/BeefBug_1783.zip)

**Comment by bfiete** [comment]
Fixed at 16de3a14a4835e05f37726c8eeb83cc9ad03c8bb

**Comment by bfiete** [comment]
Let me know if you have any issues upgrading to the next nightly build.
